api/helpers/add-game-to-match.js

```diff
@@ -64,6 +64,7 @@ module.exports = {
           winner: relevantMatch.player2,
         });
       }
+      relevantMatch = await sails.helpers.findOrCreateCurrentMatch(player1.id, player2.id);
```

Comment:
This feels like it should be in an else at first, but I'm not entirely sure. It's weird to set `relevantMatch` conditionally based on player wins above, and then reset it again immediately at the end anyway, overwriting what we did before.

If this is intentional, a comment explaining why this is necessary would be really useful.

Reply:
I believe the fundamental issue here is that the database updating operations on line 53, 56, and 62 don't ensure that the `relevantMatch()` variable is up to date with the state of the database (as generally database-updating methods don't always return the resulting object, they just update it for the next time the db is read).

@lihaobhsfer please feel free to chime in here, but I think the possibility for de-sync is threefold depending on the conditionasls on lines 56 and 61:

@lihaobhsfer am I correct that the issue you encountered was consistently that the most recently played game was not appearing in `relevantMatch()` because  `Match.addToCollection()` does not update this variable in-memory?


The thing that is most awkward about line 67 is that it re-queries the database to generate the match object that we _pretty much_ already have but not completely (because it's missing the current game).

What might be better would be to simply update `relevantMatch` in memory to reflect the expected state in the database i.e. add the current game to it. We could move the logic that adds the game to the relevant match down below the logic that updates the relevant match in case it's over (someone got 2 wins). What about something like this?

```
      // End the match if this game clinches it
      if (numPlayer1Wins >= 2) {
        relevantMatch = await Match.updateOne(relevantMatch.id).set({
          endTime: dayjs().valueOf(),
          winner: relevantMatch.player1,
        });
      } else if (numPlayer2Wins >= 2) {
        relevantMatch = await Match.updateOne(relevantMatch.id).set({
          endTime: dayjs().valueOf(),
          winner: relevantMatch.player2,
        });
      }

      // Add game to current match
     await Match.addToCollection(relevantMatch.id, 'games').members([game.id]);
     relevantMatch.games.push(game); 
```

That way we group up the database write operation with the part where we update the `relevantMatch` variable in-memory, and we avoid re-querying for data that should already be up to date. @seriouslysean and @lihaobhsfer does that make sense?

Reply:
Thanks for chiming in, this makes sense to me. I’m not as well versed on the sails side so I was relying on your breakdown, haha.

Reply:
@lihaobhsfer do you think this is reasonable and that it would solve the desync issue while avoiding an extra query?

Reply:
I'm curious to the answer above, but otherwise I'm fine with what you both decide. It might be worth adding a follow up to look at this whole chain in detail and rearchitect it, but that's not the point of this PR as long as it's working and tests well.